geode-wan-txgrouping/src/distributedTest/java/org/apache/geode/internal/cache/wan/txgrouping/serial/TxGroupingSerialDUnitTest.java

```diff
@@ -0,0 +1,366 @@
+
+    londonServer2VM.invoke(() -> checkStats_Failover(newYorkName, entries));
+  }
+
+  private void checkQueuesAreEmptyAndOnlyCompleteTransactionsAreReplicated(
+      boolean isBatchesRedistributed) {
+    // Wait for sender queues to be empty
+    List<List<Integer>> londonServersStats = new ArrayList(londonServersVM.length);
+    int i = 0;
+    for (VM londonServer : londonServersVM) {
+      londonServersStats.add(londonServer.invoke(() -> getSenderStats(newYorkName, 0)));
+    }
+
+    int queueSize = londonServersStats.stream().map(x -> x.get(0)).reduce(0, (y, z) -> y + z);
+    assertThat(queueSize).isEqualTo(0);
+
+    // batches redistributed:
+    int batchesRedistributed =
+        londonServersStats.stream().map(x -> x.get(5)).reduce(0, (y, z) -> y + z);
+    if (isBatchesRedistributed) {
+      assertThat(batchesRedistributed).isGreaterThan(0);
+    } else {
+      assertThat(batchesRedistributed).isEqualTo(0);
+    }
+  }
+
+  private void createReplicatedRegion(String regionName, String senderId) {
+    RegionFactory<Object, Object> fact =
+        cacheRule.getCache().createRegionFactory(RegionShortcut.REPLICATE);
+    if (senderId != null) {
+      fact.addGatewaySenderId(senderId);
+    }
+    fact.create(regionName);
+  }
+
+  private void checkQueueStats(String senderId, final int queueSize, final int eventsReceived,
+      final int eventsQueued, final int eventsDistributed) {
+    GatewaySenderStats statistics = getGatewaySenderStats(senderId);
+    assertThat(statistics.getEventQueueSize()).isEqualTo(queueSize);
+    assertThat(statistics.getEventsReceived()).isEqualTo(eventsReceived);
+    assertThat(statistics.getEventsQueued()).isEqualTo(eventsQueued);
+    assertThat(statistics.getEventsDistributed()).isGreaterThanOrEqualTo(eventsDistributed);
+  }
+
+  private void checkBatchStats(String senderId, final int batches,
+      final boolean batchesRedistributed) {
+    GatewaySenderStats statistics = getGatewaySenderStats(senderId);
+    assertThat(statistics.getBatchesDistributed()).isEqualTo(batches);
+
+    if (batchesRedistributed) {
+      assertThat(statistics.getBatchesRedistributed()).isGreaterThan(0);
+    } else {
+      assertThat(statistics.getBatchesRedistributed()).isEqualTo(0);
+    }
+  }
+
+  private void checkConflatedStats(String senderId) {
+    GatewaySenderStats statistics = getGatewaySenderStats(senderId);
+    assertThat(statistics.getEventsNotQueuedConflated()).isEqualTo(0);
+  }
+
+  private void checkStats_Failover(String senderId, final int eventsReceived) {
+    GatewaySenderStats statistics = getGatewaySenderStats(senderId);
+    assertThat(statistics.getEventsReceived()).isEqualTo(eventsReceived);
+    assertThat((statistics.getEventsQueued() + statistics.getUnprocessedTokensAddedByPrimary()
+        + statistics.getUnprocessedEventsRemovedByPrimary())).isEqualTo(eventsReceived);
+  }
+
+  private Boolean killPrimarySender(String senderId) {
```

Comment:
The dunit testing framework originally required primitive wrappers, so I think you may have copied some of this from elsewhere. It's better to just use simple primitives for return types and parameters now, such as `boolean` instead of `Boolean`.